src/studio/src/designer/backend/Infrastructure/GitRepository/GitRepository.cs

```diff
@@ -210,6 +211,35 @@ public bool DirectoryExitsByRelativePath(string relativeDirectoryPath)
             return Directory.Exists(absoluteDirectoryPath);
         }
 
+        /// <summary>
+        /// Copies the contents of the repository to the target directory.
+        /// </summary>
+        /// <param name="targetDirectory">Full path of the target directory</param>
+        public void CopyRepository(string targetDirectory)
+        {
```

Comment:
Should have a guard here to make sure the targetDirectory isn't outside the repositories root directory. Guard.AssertFilePathWithinParentDirectory(RepositoryDirectory, targetDirectory);